react/react1/week3/to-do/src/components/FetchData.jsx

```diff
@@ -0,0 +1,104 @@
+import { useEffect, useState } from "react";
+import "react-datepicker/dist/react-datepicker.css";
+import Header from "./Header";
+import TodoItem from "./ToDoItem";
+import DatePicker from "react-datepicker";
+const FetchData = () => {
+  const [todos, setToDos] = useState([]);
+  const [edit, setEEdit] = useState(false);
+  const [newToDo, setNewToDo] = useState({});
+  const [deadLine, setDeadLine] = useState(new Date());
+  const NewToDo = (description) => {
+    setNewToDo({
+      id: Math.floor(Math.random() * (100 - 4 + 1) + 4),
+      description: description,
+      deadline: deadLine,
+      done: false,
+    });
+  };
+  const AddToDo = () => {
+    setToDos((previous) => [...previous, newToDo]);
+  };
+  const DeleteToDo = (id) => {
+    setToDos(todos.filter((eachToDo) => eachToDo.id !== id));
+  };
+  const editMode = (id) => {
+    setEEdit(!edit);
+    const index = todos.indexOf(todos.find((eachTodo) => eachTodo.id === id));
+    if (index !== -1) todos[index].isEdit = !todos[index].isEdit;
+    setToDos(todos);
+  };
+  const handleDesc = (desc, id) => {
+    todos.forEach((todo) =>
+      todo.id === id ? (todo.description = desc) : todo.description
+    );
+    const index = todos.indexOf(todos.find((eachTodo) => eachTodo.id === id));
+    if (index !== -1) todos[index].isEdit = !todos[index].isEdit;
+    setToDos(todos);
+    setEEdit(!edit);
+  };
+  useEffect(() => {
+    (async () => {
+      const data = await fetch(
+        "https://gist.githubusercontent.com/benna100/391eee7a119b50bd2c5960ab51622532/raw"
+      );
+      const result = await data.json();
+      console.log(result);
+      setToDos(result);
+    })();
+  }, []);
+
+  return (
+    <div>
+      <Header />
+      <div className="add">
+        <label className="label"> To Do Description:</label>
+        <input
+          type="text"
+          onChange={(e) => NewToDo(e.target.value)}
+          placeholder="Enter the To-do you want to add"
+        ></input>
+        <br />
+        <label className="label">Deadline:</label>
+        <DatePicker
+          showIcon
+          dateFormat="yyyy/MM/dd"
+          selected={deadLine}
+          onChange={(date) => {
+            date < new Date() ? alert("Invalid deadline!") : setDeadLine(date);
+            setDeadLine(date);
+          }}
+        />
```

Comment:
I thought `FetchData `file is to fetch Data, but you are rendering too! So name doesn't make sense here!